### PR TITLE
Consensus code cleanup

### PR DESCRIPTION
**Pull request by tvolk131**

### CODE REVIEW

File: fedimint-server/src/consensus/mod.rs [hunk 1]
```diff
@@ -424,7 +424,9 @@ impl FedimintConsensus {
         current
     }
 
+    /// Wait for any module to send us a proposal
```
Comment (elsirion) [inline fedimint-server/src/consensus/mod.rs:427]: @joschisan will remove this method soon. Might be worth dropping the commit to save him some rebase conflicts.

File: fedimint-server/src/consensus/server.rs [hunk 2]
```diff
@@ -442,7 +441,7 @@ impl ConsensusServer {
                 _ => break vec![],
             };
         };
-        let proposal = self.process_events_then_propose(override_proposal).await;
```
Comment (elsirion) [inline fedimint-server/src/consensus/server.rs:461]: On mobile, so can't test, but why doesn't this break the functionality that we can force a guardian to accept a specific epoch? See #2309

**Review by elsirion — COMMENTED** [review]
Nicely structured PR!

File: fedimint-server/src/consensus/mod.rs (hunk 1 above)
Comment (tvolk131) [reply fedimint-server/src/consensus/mod.rs]: Done.

File: fedimint-server/src/consensus/server.rs (hunk 2 above)
Comment (tvolk131) [reply fedimint-server/src/consensus/server.rs:461]: I'm not familiar enough with the consensus code to have much input on that. I just noticed that the only place `override_proposal` was used, `None` was passed in. If there are plans to eventually use this for something, I can drop the commit!

Comment (dpc) [reply fedimint-server/src/consensus/server.rs:461]: I think it's a leftover of forcing an epoch being moved to be admin api endpoint.

```
    // Save any API events we have in the channel then create a proposal
    async fn process_events_then_propose(
        &mut self,
        override_proposal: Option<ConsensusProposal>,
    ) -> ConsensusProposal {
        while let Some(Some(event)) = self.api_receiver.next().now_or_never() {
            match event {
                ApiEvent::ForceProcessOutcome(outcome) => self.force_process_epoch(outcome).await,
                event => {
                    self.consensus.api_event_cache.insert(event);
                }
            };
        }
        let consensus_proposal = self.consensus.get_consensus_proposal().await;
        self.consensus.api_event_cache.clear();
        override_proposal.unwrap_or(consensus_proposal)
    }
```

We now force an epoch as an event sent through `api_endpoint`, while the `override_proposal: Option<ConsensusProposal>,` never actually gets set to any `Some` on the server side. @elsirion

**Review by dpc — APPROVED** [review]

**Review by elsirion — APPROVED** [review]
Some good improvements!

**Pull request merged**